xds/src/test/java/io/grpc/xds/XdsServerTestHelper.java

```diff
@@ -230,27 +248,61 @@ public <T extends ResourceUpdate> void cancelXdsResourceWatch(XdsResourceType<T>
     }
 
     @Override
-    public void shutdown() {
+    public synchronized void shutdown() {
       shutdown = true;
     }
 
     @Override
-    public boolean isShutDown() {
+    public synchronized boolean isShutDown() {
       return shutdown;
     }
 
+    public void awaitRds() throws InterruptedException, TimeoutException {
+      if (!rdsCount.await(WAIT_TIMEOUT.toMillis(), TimeUnit.MILLISECONDS)) {
```

Comment:
Pass the timeout as arguments to `awaitRds()` so it is obvious in the test what is happening and can be adjusted.